src/Raven.Server/Documents/PeriodicBackup/Restore/RestoreBackupTaskBase.cs

```diff
@@ -352,6 +352,45 @@ public async Task<IOperationResult> Execute(Action<IOperationProgress> onProgres
             }
         }
 
+        private async Task SaveDatabaseRecordAsync(string databaseName, DatabaseRecord databaseRecord, Dictionary<string, 
+            BlittableJsonReaderObject> databaseValues, RestoreResult restoreResult, Action<IOperationProgress> onProgress)
+        {
+            const int maxRetries = 10;
+            var retries = 0;
+            long index;
+
+            // at this point we restored a large portion of the database or all of it
+            // we'll retry saving the database record as failure here will cause us to abort the whole restore operation
+
+            while(true)
+            {
+                try
+                {
+                    _operationCancelToken.Token.ThrowIfCancellationRequested();
+
+                    restoreResult.AddInfo("Saving the database record");
+                    onProgress.Invoke(restoreResult.Progress);
+
+                    var result = await _serverStore.WriteDatabaseRecordAsync(
+                        databaseName, databaseRecord, null, RaftIdGenerator.NewId(), databaseValues, isRestore: true);
+                    index = result.Index;
+                    break;
+                }
+                catch (TimeoutException)
+                {
+                    if (++retries < maxRetries)
+                        continue;
+
+                    restoreResult.AddError("Failed to save the database record, the restore is aborted");
+                    onProgress.Invoke(restoreResult.Progress);
+                    throw;
+                }
+            }
+
+            // we are waiting for the command to propagate to this node after it was accepted by the leader
+            await _serverStore.Cluster.WaitForIndexNotification(index, TimeSpan.FromSeconds(60));
```

Comment:
Reaching here, doesn't ensure that the command will be committed.
This ensures that it is in the the leader's log _only_, but it has to be in the majority of the nodes in order to be committed.

Reply:
I would move this into the `try/catch`

Reply:
Maybe it should be in its own `try/catch`?
What if `WriteDatabaseRecordAsync` succeeds and we only fail in `WaitForIndexNotification`?